server.go

```diff
@@ -172,22 +171,20 @@ func (s *Server) Boot() error {
 func (s *Server) goAcceptConnection(listener net.Listener) {
 	s.wait.Add(1)
 	go func(listener net.Listener) {
-	loop:
+		defer s.wait.Done()
 		for {
 			select {
-			case <-s.doneTcp:
-				break loop
+			case <-s.stop:
+				return
 			default:
-			}
-			connection, err := listener.Accept()
-			if err != nil {
-				continue
-			}
+				connection, err := listener.Accept()
+				if err != nil {
+					continue
+				}
 
-			s.goScanConnection(connection)
+				s.goScanConnection(connection)
+			}
 		}
-
-		s.wait.Done()
```

Comment:
They decremented the waitgroup outside of the for loop only for this function. Inconsistent indeed